src/charm.py

```diff
@@ -283,6 +283,40 @@ def _spring_boot_env(self) -> typing.Dict[str, str]:
             env[JAVA_TOOL_OPTIONS] = jvm_config
         return env
 
+    def _detect_mysql_capability(self) -> bool:
+        """Detect if the mysql lib is present in the spring-boot app.
+
+        Returns:
+            True if the mysql lib is present, False otherwise.
+
+        Raises:
+            TypeError: If the spring-boot app has a not recognizable type.
+        """
+        java_app = self._detect_java_application()
+        container = self._spring_boot_container()
+
+        if isinstance(java_app, BuildpackApplication):
+            try:
+                files = container.list_files("/workspace/BOOT-INF/lib")
+            except FileNotFoundError:
+                return False
+
+            return any(
+                file.name.startswith("mysql-connector-j-") and file.name.endswith(".jar")
```

Comment:
Can you refactor this so that `file.name.startswith("mysql-connector-j-") and file.name.endswith(".jar")`is not repeated here and in line 314?

Reply:
I decided to not use the pattern matching to avoid having to mock it for unit tests. There's already a lot of mocking going on in these unit tests.  
But thank you for mentioning it.